### PR TITLE
Update names of parameter for SVM

### PR DESCRIPTION
**Pull request by PetrovKP**
Renamed next params:
```c++
k->scale
b->shift
coefficients->coeffs
```

### CODE REVIEW

**Comment by PetrovKP** [comment]
@rlnx , @michael-smirnov please, review this changes

**Comment by PetrovKP** [comment]
I will merge with master, after review or approve.

File: cpp/oneapi/dal/algo/linear_kernel/common.cpp
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/linear_kernel/common.cpp:39]: In my point of view, if `descriptor_base` is in public namespace (`oneapi::dal::linear_kernel`), when user potentially may use it and we have to have a docs for it. Then, name setters `set_something_impl()` is not appropriate in this case, since it's public API. Moreover, it is inconsistent with getter names: (`set_scale_impl()` but `get_scale()` in `descriptor base`). I think we can move descriptor_base to details and name all of its methods consistently, with *_impl* or without it. It's how it done in data management - please see `table`, `homogen_table`, `row_accessor` for example.

File: cpp/oneapi/dal/algo/svm/backend/cpu/infer_kernel.cpp
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/svm/backend/cpu/infer_kernel.cpp:52]: Can we introduce some simple rule to shorten the names? We have `get_coeffs()` in the model but `get_column_count()` in tables, which can be shorten also to `get_col_count()`

File: cpp/oneapi/dal/algo/svm/detail/infer_ops.hpp
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/svm/detail/infer_ops.hpp:49]: nip: I think we shouldn't shorten this name in exception descriptions

File: cpp/oneapi/dal/algo/svm/detail/infer_ops.hpp
Comment (michael-smirnov) [inline cpp/oneapi/dal/algo/svm/detail/infer_ops.hpp:64]:
```suggestion
                "Row count of coefficients should be equal to support vectors count in the input model");
```

File: examples/oneapi/cpp/source/svm/svm_two_class_thunder_dense_batch.cpp
Comment (michael-smirnov) [inline examples/oneapi/cpp/source/svm/svm_two_class_thunder_dense_batch.cpp:36]: Please merge master, there are API changes

File: cpp/oneapi/dal/algo/svm/backend/cpu/infer_kernel.cpp
Comment (PetrovKP) [reply cpp/oneapi/dal/algo/svm/backend/cpu/infer_kernel.cpp:52]: We have to negotiate within the library. Coefficients have been cut due to the fact that the graphs have a approved name `coeff`. (https://github.com/oneapi-src/oneDAL/pull/823/files#diff-38367a019750050cdad24bcdde40989aR94)

File: cpp/oneapi/dal/algo/linear_kernel/common.cpp
Comment (rlnx) [inline cpp/oneapi/dal/algo/linear_kernel/common.cpp:39]: Can we move it to protected?

File: cpp/oneapi/dal/algo/svm/detail/infer_ops.hpp
Comment (PetrovKP) [reply cpp/oneapi/dal/algo/svm/detail/infer_ops.hpp:64]: We had a rule in this order to draw up exceptions. @mchernov-intel Correct me if I'm wrong.

File: cpp/oneapi/dal/algo/linear_kernel/common.cpp
Comment (michael-smirnov) [reply cpp/oneapi/dal/algo/linear_kernel/common.cpp:39]: It solves only part of the problem - `descriptor_base` is still in public interfaces. Do we assume that it can be used by customers? I was thinking that it is a service functionality and is a part of our internal architecture

**Review by michael-smirnov — APPROVED** [review]

**Pull request merged**
